File: WareHouseManagement Mobile/warehouse_mobile/lib/models/supplier.dart
class Supplier {
  final String id;
  final String name;
  final String country;
  final String adress;
  final String email;
  final String phone;

  Supplier({
    required this.id,
    required this.name,
    required this.country,
    required this.adress,
    required this.email,
    required this.phone,
  });

  factory Supplier.fromJson(Map<String, dynamic> json) {
    return Supplier(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      country: json['country'] ?? '',
      adress: json['adress'] ?? '',
      email: json['email'] ?? '',
      phone: json['phone'] ?? '',
    );
  }
}

File: WareHouseManagement API/Domain/Entities/Supplier.cs
using Domain.BaseEntities;

namespace Domain.Entities;

public class Supplier:BaseEntity
{
    public string Country { get; set; }
    public string Adress { get; set; }
    public string Email { get; set; }
    public string Phone { get; set; }
    public ICollection<Product> Products { get; set; } = new List<Product>();
    public ICollection<Order> Orders { get; set; } = new List<Order>();

}
